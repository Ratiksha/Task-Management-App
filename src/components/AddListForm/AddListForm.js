import React, { Component } from 'react';
import './AddListForm.scss';

export default class AddListForm extends Component{
    constructor(props){
        super(props);
        this.state = {
            value: '',
            errorRequired: false,
            errorDuplicate: false,
            disableSubmit: true
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange = (event) => {
        this.setState({
            value: event.target.value, 
            errorRequired:false, 
            errorDuplicate: false,
            disableSubmit: false
        });
    }
    handleSubmit = (event) => {
        const { addList, listId, editList, hideModal } = this.props
        const { value } = this.state
        event.preventDefault();
        if(!this.checkDuplicate(value)){
            if(value && editList){
                editList(value, listId);
                hideModal();
            }
            else if (value && addList) {
                addList(value);
                hideModal();
            }
            else{
                this.setState({
                    errorRequired: true
                })
            }
        }
        
    }

    checkDuplicate(name){
        const { lists } = this.props
        for (var i = 0; i < lists && lists.length; i++) {
            if(name.toUpperCase() === lists[i].name.toUpperCase()){
                this.setState({
                    errorDuplicate: true
                })
                return true
            }
          }
          
    }

    
    render() {
        const { value, errorRequired, errorDuplicate, disableSubmit } = this.state;
        const { addList, editList, name } = this.props;
        return (
            <form onSubmit={this.handleSubmit}>
                {addList ? <h3>Add Task List</h3> : <h3>Edit Task List</h3> }
                <input type="text" defaultValue={name? name : value} onChange={this.handleChange} className={`${errorRequired || errorDuplicate? 'highlight': ''}`} />
                <input type="submit" value="Submit" className="button green" disabled={editList ? disableSubmit : false} />
                {errorRequired && <p className="error">Please enter the name</p>}
                {errorDuplicate && <p className="error">Task List already exists</p>}
            </form>
        )
    }
}