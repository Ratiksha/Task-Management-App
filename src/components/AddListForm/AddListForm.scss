.error{
    font-size: 12px;
    margin-top: 0;
    color: red
}

.highlight{
    border: 1px solid red;
}

.submit-button{
    margin-left: 10px;
}