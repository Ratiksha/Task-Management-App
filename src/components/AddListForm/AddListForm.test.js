import React from 'react';
import { shallow } from 'enzyme';
import AddListForm from './AddListForm';

let wrapped = shallow(<AddListForm />);
describe('AddListForm', () => {
  it('should render the AddListForm Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('When submit button is clicked!', () => {
    const preventDefault = jest.fn();
    wrapped.find('form').simulate('submit', { preventDefault })
    expect(preventDefault).toBeCalled();
  });

  it('When input is clicked', () => {
    const mockedEvent = { target: { value: "foo"} }
    const spy = jest.spyOn(wrapped.instance(), 'handleChange')
    wrapped.instance().forceUpdate();
    wrapped.find('input[type="text"]').simulate('change', mockedEvent)
    expect(spy).toHaveBeenCalledTimes(1);
  })
});