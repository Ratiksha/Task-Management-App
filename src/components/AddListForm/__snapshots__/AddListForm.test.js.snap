// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddListForm should render the AddListForm Component correctly 1`] = `ShallowWrapper {}`;
